protostar/commands/test/test_runner.py

```diff
@@ -176,8 +172,10 @@ async def _run_test_suite(
                         file_path=test_suite.test_path,
                         test_case_name=test_case_name,
                         exception=ex,
-                        captured_setup_stdout=setup_stdout_buffer.getvalue(),
-                        captured_test_stdout=test_stdout_buffer.getvalue(),
+                        captured_stdout={
+                            key: value.getvalue()
+                            for key, value in new_execution_state.output_recorder.captures
+                        },
```

Comment:
this dict building might be worth encapsulating as


```suggestion
                        captured_stdout=new_execution_state.output_recorder.get_captures(),
```